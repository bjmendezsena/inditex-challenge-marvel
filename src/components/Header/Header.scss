@import "@/styles/variables.scss";

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2px 48px;
  height: $header-height;
  background-color: $secondary-color;

  @media screen and (max-width: $mobile-width) {
    padding: 0 18px;
  }

  &__favourite {
    color: $white-color;
    justify-content: center;
    align-items: center;
    display: flex;
  }
}
