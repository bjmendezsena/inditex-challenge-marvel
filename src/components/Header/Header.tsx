import React from "react";
import Image from "next/image";
import Link from "next/link";
import { FavButton } from "@/components";
import { useFavourites } from "@/context";
import "./Header.scss";
export const Header = () => {
  const { favourites, setShowFavourites, showFavourites } = useFavourites();

  const handleShowFavourites = () => {
    if (showFavourites || !favourites.length) return;
    setShowFavourites(true);
  };

  return (
    <div className='header'>
      <Link href={"/"}>
        <Image
          onClick={() => {
            setShowFavourites(false);
          }}
          src='/logo.svg'
          alt='logo'
          width={130}
          height={52}
        />
      </Link>
      <div className='header__favourite'>
        <FavButton size='xl' onClick={handleShowFavourites} isActive />{" "}
        <p>{favourites.length}</p>
      </div>
    </div>
  );
};
