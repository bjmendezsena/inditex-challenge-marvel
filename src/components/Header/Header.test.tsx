import { render } from "@testing-library/react";
import { Header } from "./Header";

describe(`<${Header.name} />`, () => {
  it("should render without crashing", () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });
});
