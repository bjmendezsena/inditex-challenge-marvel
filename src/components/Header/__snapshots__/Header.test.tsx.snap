// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <a
      href="/"
    >
      <img
        alt="logo"
        data-nimg="1"
        decoding="async"
        height="52"
        loading="lazy"
        src="/logo.svg"
        style="color: transparent;"
        width="130"
      />
    </a>
    <div
      class="header__favourite"
    >
      <button
        class="FavButton FavButton--active"
      >
        <svg
          class="icon icon__size--xl"
          fill="none"
          viewBox="0 0 56 38"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
       
      <p>
        0
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Header /> should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <img
      alt="logo"
      data-nimg="1"
      decoding="async"
      height="52"
      loading="lazy"
      src="/logo.svg"
      style="color: transparent;"
      width="130"
    />
    <div
      class="header__favourite"
    >
      <button
        class="FavButton FavButton--active"
      >
        <svg
          class="icon icon__size--xl"
          fill="none"
          viewBox="0 0 56 38"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
       3
    </div>
  </div>
</DocumentFragment>
`;
